sklearn/cluster/optics_.py

```diff
@@ -0,0 +1,774 @@
+        node. A lower value for the similarity threshold means less levels
+        in the tree.
+
+    significant_min : float, optional
+        Sets a lower threshold on how small a significant maxima can be.
+
+    min_cluster_size_ratio : float, optional
+        Minimum percentage of dataset expected for cluster membership.
+
+    min_maxima_ratio : float, optional
+        Used to determine neighborhood size for minimum cluster membership.
+
+    algorithm : {'auto', 'ball_tree', 'kd_tree', 'brute'}, optional
+        Algorithm used to compute the nearest neighbors:
+        - 'ball_tree' will use :class:`BallTree`
+        - 'kd_tree' will use :class:`KDTree`
+        - 'brute' will use a brute-force search.
+        - 'auto' will attempt to decide the most appropriate algorithm
+          based on the values passed to :meth:`fit` method.
+
+        Note: fitting on sparse input will override the setting of
+        this parameter, using brute force.
+
+    leaf_size : int, optional (default=30)
+        Leaf size passed to :class:`BallTree` or :class:`KDTree`. This can
+        affect the speed of the construction and query, as well as the memory
+        required to store the tree. The optimal value depends on the
+        nature of the problem.
+
+    n_jobs : int, optional (default=1)
+        The number of parallel jobs to run for neighbors search.
+        If ``-1``, then the number of jobs is set to the number of CPU cores.
+        Affects only :meth:`kneighbors` and :meth:`kneighbors_graph` methods.
+
+    Attributes
+    ----------
+    core_sample_indices_ : array, shape (n_core_samples,)
+        Indices of core samples.
+
+    labels_ : array, shape (n_samples,)
+        Cluster labels for each point in the dataset given to fit().
+        Noisy samples are given the label -1.
+
+    reachability_ : array, shape (n_samples,)
+        Reachability distances per sample.
+
+    ordering_ : array, shape (n_samples,)
+        The cluster ordered list of sample indices
+
+    core_distances_ : array, shape (n_samples,)
+        Distance at which each sample becomes a core point.
+        Points which will never be core have a distance of inf.
+
+    See also
+    --------
+
+    DBSCAN
+        CPU optimized algorithm that clusters at specified neighborhood
+        radius (eps).
+    HDBSCAN
+        Related clustering algorithm that calculates the minimum spanning tree
+        across mutual reachability space.
+
+    References
+    ----------
+    Ankerst, Mihael, Markus M. Breunig, Hans-Peter Kriegel, and Jörg Sander.
+    "OPTICS: ordering points to identify the clustering structure." ACM SIGMOD
+    Record 28, no. 2 (1999): 49-60.
+    """
+
+    def __init__(self, min_samples=5, max_bound=np.inf, metric='euclidean',
+                 p=2, metric_params=None, maxima_ratio=.75,
+                 rejection_ratio=.7, similarity_threshold=0.4,
+                 significant_min=.003, min_cluster_size_ratio=.005,
```

Comment:
How did you set the default `significant_min=.003`? It is homogenous to a distance, isn't it? Should it be normalized in some way?

Reply:
...that's a very good point. The 'auto' extraction function was forked and modified from here: https://github.com/amyxzhang/OPTICS-Automatic-Clustering/blob/master/AutomaticClustering.py 
...and the default of 0.003 was inherited from that codebase. I absolutely agree that it needs to be normalized, since the magnitude of the input units shouldn't have bearing on the extraction. I'm not exactly sure the best way to do this, but ideally the user should be able to identify the `significant_min` parameter as a percentage. Perhaps if the reachability distances are normalized so the largest distance is '1'? It would have to be only in this extraction function though-- calling the dbscan() method expects an epsilon in original units (i.e., the reachability graph that's stored shouldn't be modified).

I've gone ahead and normalized as I described above; this shouldn't impact the other parameters as they all are either ratios or percents...